File: backend/src/main/java/com/cachemeifyoucan/econometro/application/dto/AlertResponse.java
package com.cachemeifyoucan.econometro.application.dto;

import java.math.BigDecimal;

public record AlertResponse( 
    long productId,
    String productTitle,
    BigDecimal desiredPrice,
    BigDecimal bestPrice,
    String imageContent,
    Offer fullfillingOffer
) {

     public static record Offer(long sellerId, String sellerName, String sellerImage, BigDecimal price) {

    }

}

File: frontend/src/app/domain/dto/alert-response.ts
export interface AlertResponse {
  productId: number;
  productTitle: string;
  desiredPrice: number;
  bestPrice: number;
  imageContent: string;
  fullfillingOffer: {
    sellerId: number;
    sellerName: string;
    sellerImage: string;
    price: number;
  };
}
